pkg/agent/monitortool/latency_store_test.go

```diff
@@ -0,0 +1,245 @@
+			prevExpectedEntry: NodeIPLatencyEntry{},
+			expectedEntry:     NodeIPLatencyEntry{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.key, func(t *testing.T) {
+			latencyStore := &LatencyStore{
+				isNetworkPolicyOnly: false,
+				nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+
+			entry, _ := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, tt.prevExpectedEntry, entry)
+			latencyStore.DeleteNodeIPLatencyEntry(tt.key)
+			entry, ok := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, entry, tt.expectedEntry)
+			assert.False(t, ok)
+		})
+	}
+}
+
+func TestLatencyStore_SetNodeIPLatencyEntry(t *testing.T) {
+	tests := []struct {
+		key           string
+		updatedEntry  NodeIPLatencyEntry
+		expectedEntry NodeIPLatencyEntry
+	}{
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry,
+			expectedEntry: entry,
+		},
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry2,
+			expectedEntry: entry2,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.key, func(t *testing.T) {
+			latencyStore := &LatencyStore{
+				isNetworkPolicyOnly: false,
+				nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+
+			mutator := func(entry *NodeIPLatencyEntry) {
+				entry.LastSendTime = tt.updatedEntry.LastSendTime
+				entry.LastRecvTime = tt.updatedEntry.LastRecvTime
+				entry.LastMeasuredRTT = tt.updatedEntry.LastMeasuredRTT
+			}
+			latencyStore.SetNodeIPLatencyEntry(tt.key, mutator)
+			entry, ok := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, tt.expectedEntry, entry)
+			assert.True(t, ok)
+		})
+	}
+}
+
+func TestLatencyStore_DeleteStaleNodeIPs(t *testing.T) {
+	tests := []struct {
+		key           string
+		updatedEntry  NodeIPLatencyEntry
+		expectedEntry NodeIPLatencyEntry
+	}{
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry,
+			expectedEntry: entry,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.key, func(t *testing.T) {
+			latencyStore := &LatencyStore{
+				isNetworkPolicyOnly: false,
+				nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+			// Remove node
+			latencyStore.deleteNode(&corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node1",
+				},
+			})
+
+			// Check if the entry is deleted
+			_, ok := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.False(t, ok)
+
+			// Send the echo from the node
+			mutator := func(entry *NodeIPLatencyEntry) {
+				entry.LastSendTime = tt.updatedEntry.LastSendTime
+				entry.LastRecvTime = tt.updatedEntry.LastRecvTime
+				entry.LastMeasuredRTT = tt.updatedEntry.LastMeasuredRTT
```

Comment:
it seems that you could use an empty mutator here to make the test less verbose, as it doesn't impact the validity of the test

Reply:
OK, I have changed it to empty.